core/src/main/scala/loadbalancers/Aws.scala

```diff
@@ -97,35 +92,82 @@ final class Aws(cfg: Infrastructure.Aws) extends (LoadbalancerOp ~> IO) {
   // The target groups have a restriction such that name of the target group can't
   // have "double dashes" and the entire name must be less than 32 characters.
   // Unfortunately, target groups will not have the same name as the loadbalancers.
-  // An example target group will be named stack-1-0-0-443 where the last number is the
-  // port that it listens on (instead of the hash). This allows stack-1-0-0-80 to be
+  // An example target group will be named stack-aevaif-443 where the last number is the
+  // port that it listens on (instead of the hash). This allows stack-qweri-80 to be
   // a different target group for the same loadbalancer. Since target groups are only
   // associated with a single loadbalancer at a time, it'll be easy to reason about which target
-  // groups are related to which loadbalancers.
-  def targetGroupNamePrefix(name: String, v: MajorVersion) =
-    s"$name-${v.minVersion.toExternalString}"
+  // groups are related to which loadbalancers. Administrators will need to reason about
+  // which target group belongs to which loadbalancer based on the hash in the target group name.
+  def targetGroupNamePrefix(name: String, hash: String) =
+    s"$name-$hash"
 
-  def getLoadBalancerArn(name: String): IO[String] = {
+  def getLoadBalancerArn(name: String): IO[Option[String]] = {
     val describeLoadBalancerRequest = new DescribeLoadBalancersRequest().withNames(name)
     (for {
       r <- IO(cfg.nlb.describeLoadBalancers(describeLoadBalancerRequest))
       x = r.getLoadBalancers.asScala.toList
       a = x.head.getLoadBalancerArn
-    } yield a).recoverWith{
+    } yield Option(a)).recoverWith{
       case e: Exception =>
-        log.error(s"aws call to get loadbalancer ARN ${describeLoadBalancerRequest.getNames.asScala.head} failed.")
-        IO.raiseError(FailedLoadbalancerDeploy(s"could not find loadbalancer and therefore could not delete loadbalancer", e.getMessage))
+        log.error(s"aws call to get loadbalancer ARN ${describeLoadBalancerRequest.getNames.asScala.head} failed with ${e.getMessage}")
+        IO(None)
```

Comment:
Does it semantically make sense to have an `Option` return here? It's unclear to me what successful result scenario would be `None`.

Reply:
Yeah I have no idea. If the network call is successful, then I want to return the string. If it's not successful, I'm not sure what to return.

In my mind, I was thinking that I want to return _something_ if the call is successful and _nothing_ if it's not. `Option` seemed like the way to go. What did you have in mind?

Reply:
In this case, one would not usually leverage `Option`; here we would use `Either` where the result is _either_ a `String` or `Error`, but the resulting type is clear in either branch (semantically, and practically). In essence, this is what you're getting with `IO` - your computation either succeeds or it doesn't, but the semantics of IO are specifically for IO-related things (which you have in this case).

Does that help?